### PR TITLE
fix: add raspberry pi 3B wifi drivers

### PR DESCRIPTION
**Pull request by Neonox31**
This P.R adds wifi drivers for the raspberry pi 3B+ board.

https://github.com/sgielen/picl-k3os-image-generator/issues/6

### CODE REVIEW

**Comment by Roflincopter** [comment]
I Tested this as well on an rpi-4 and it works, however after I had to prepend sudo to all of your commands. I think you tested by running the entire script as root? which would mean the dependencies are also downloaded as root?

In any case to be more consistent with the rest of the script I would propose prepending sudo to your mkdir and cp's.

**Comment by Neonox31** [comment]
@Roflincopter You're right, this is now fixed 👍

**Comment by t-h-a-l-i-s** [comment]
> @Roflincopter You're right, this is now fixed 👍

Thanks for that fix. But you have to look at the `dmesg` output which firmware is missing for your device. For my RPi Model 3B+ I had to load the `brcmfmac43430-sdio*` files.

**Comment by Neonox31** [comment]
OK I will add 43430 drivers too.

**Comment by Neonox31** [comment]
OK now 43430 drivers are also included and `7z` command is used instead of `unzip` one.

File: build-image.sh [hunk 1]
```diff
@@ -267,6 +268,11 @@ if [ "$IMAGE_TYPE" = "orangepipc2" ]; then
 	sudo ln -s $(cd root/boot; ls -d dtb-*-sunxi64 | head -n1) root/boot/dtb
 	unpack_deb "linux-image-dev-sunxi64.deb" "root"
 	sudo ln -s $(cd root/boot; ls -d vmlinuz-*-sunxi64 | head -n1) root/boot/Image
+elif [ "$IMAGE_TYPE" = "raspberrypi" ]; then
+  sudo 7z e -y deps/rpi-firmware-nonfree-master.zip -odeps/brcm "firmware-nonfree-master/brcm/*" > /dev/null
```
Comment (mcanevet) [inline build-image.sh:272]: I think you should not invoke 7z as root, only the other commands.

File: build-image.sh [hunk 2]
```diff
@@ -267,6 +268,11 @@ if [ "$IMAGE_TYPE" = "orangepipc2" ]; then
...
+  sudo mkdir -p root/lib/firmware/brcm/
+  sudo cp deps/brcm/brcmfmac43455* root/lib/firmware/brcm/
+  sudo cp deps/brcm/brcmfmac43430* root/lib/firmware/brcm/
```
[the remark this answers is not in the document]
Comment (Neonox31) [reply build-image.sh]: Thanks, fixed !

File: build-image.sh (hunk 1 above)
Comment (Neonox31) [reply build-image.sh]: Thanks for your note, fixed.

**Comment by sgielen** [comment]
Thank you @Neonox31!

**Pull request merged**